### ISSUE TITLE
Ability to layer profiles

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#19181]
As mentioned also in: https://github-com/microsoft/vscode/-/15909-issuecomment-1464172862

The release of Profiles feature is exciting but I can't figure out how/if I can use this for "mode switching" in my own dev experience.

I'm ~~starting to~~ _hoping to_ use Profiles to keep my IDE more lean and focused - for instance one profile for **Python** development, one for **Infra** development, and one for **Data Analysis**. But what I'm finding is that the isolation between profiles is a bit _too_ strong in cases. Perfect example is my use of spell check with [this fantastic extension](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) - I have built a habit of always adding exceptions/additions to the default spell check dictionary to my user settings - these follow me everywhere. But now with profiles I think they won't anymore?

I don't know if we're going to get updates to Profiles feature that will allow some settings to follow us across profiles. The example above for the spell check dictionary, but also applies for extension settings like, [ChatGPT](https://marketplace.visualstudio.com/items?itemName=gencay.vscode-chatgpt), [Live Share](https://marketplace.visualstudio.com/items?itemName=MS-vsliveshare.vsliveshare), etc. 

Now with profiles, I think I'm going to end up repeating the same changes, and having to deal with drift across these profiles when there are some settings I just want to have permanent.

-- Proposal

Basically, there are certain plugins I'd like to treat as "global" or "personal" and those would ideally have these behaviors:

- Their settings should be maintained in my global user profile, and would not be duplicated for each of my "modal" profiles.
- When I export my profile to share with others, probably I am trying to export my modal profile settings, and not my personal/global ones.
- I think this basically requires two layers of profiles to be active at the same time: the "base" default/global/personal profile, and the "applied" profile like "Python", "Terraform", etc. - which are focused on what kind of work I'm trying to get done in the moment.

------------

Is there an option to do something like this today? It is being considered for future development?
